pkg/v1/tkg/client/client_helper.go

```diff
@@ -0,0 +1,113 @@
+// Copyright 2021 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package client
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+	"golang.org/x/sync/errgroup"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	kapppkgv1alpha1 "github.com/vmware-tanzu/carvel-kapp-controller/pkg/apiserver/apis/datapackaging/v1alpha1"
+	runtanzuv1alpha3 "github.com/vmware-tanzu/tanzu-framework/apis/run/v1alpha3"
+	"github.com/vmware-tanzu/tanzu-framework/pkg/v1/tkg/clusterclient"
+	"github.com/vmware-tanzu/tanzu-framework/pkg/v1/tkg/log"
+)
+
+// GetClusterBootstrap returns ClusterBootstrap object for the given clustername in the management cluster
+func GetClusterBootstrap(managementClusterClient clusterclient.Client, clusterName, namespace string) (*runtanzuv1alpha3.ClusterBootstrap, error) {
+	clusterBootstrap := &runtanzuv1alpha3.ClusterBootstrap{}
+	err := managementClusterClient.GetResource(clusterBootstrap, clusterName, namespace, nil, &clusterclient.PollOptions{Interval: clusterclient.CheckResourceInterval, Timeout: clusterclient.PackageInstallTimeout})
+	return clusterBootstrap, err
+}
+
+// GetCorePackagesFromCB returns addon's core packages details from the given ClsuterBootstrap object
+func GetCorePackagesFromCB(clusterBootstrap *runtanzuv1alpha3.ClusterBootstrap, namespace string) []kapppkgv1alpha1.Package {
+	var packages []kapppkgv1alpha1.Package
+	if clusterBootstrap.Spec.Kapp != nil {
```

Comment:
The code inside all the `if` blocks are same. They can be converted to a single function and parameterized by the package name such as `clusterBootstrap.Spec.CNI.RefName`. This will avoid code duplication.

Reply:
PR was not ready to review, it is ready now you can review it again.